### PR TITLE
🐛 Fix log level

### LINKED ISSUE #5736: clusterctl on main seems to not respect anymore the log level
**What steps did you take and what happened:**
run clusterctl commands, eg. `clusterctl describe cluster` and got some unexpected log lines (No default config file available)

```
No default config file available
...
No default config file available
⚠️  Using a development build of clusterctl. CLIVersion="1.0.0-315-8a5e53e09ad0a5-dirty" LatestGithubRelease="v1.0.1"
```

It seems that somehow, clustectl now does not respect anymore the log level

**What did you expect to happen:**
clusterctl to print additional log lines only when -v flag is used

**Environment:**

- Cluster-api version: main

/kind bug
/area clusterctl

/milestone v1.1

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5736

### CODE REVIEW

**Comment by stmcginnis** [comment]
/lgtm

Thanks for the test coverage.

File: cmd/clusterctl/log/logger_test.go [hunk 1]
```diff
@@ -104,3 +105,40 @@ func TestFlatten(t *testing.T) {
 		})
 	}
 }
+
+func Test_logger_Enabled(t *testing.T) {
```
Comment (schrej) [inline cmd/clusterctl/log/logger_test.go:109]: nit: not sure if there is a naming convention on casing, but I'd call this `TestLoggerEnabled`

Comment (mkumatag) [reply cmd/clusterctl/log/logger_test.go:109]: This is just an autogenerated function name from the goland editor, let me rename it to the mentioned one to align with the existing code

Comment (mkumatag) [reply cmd/clusterctl/log/logger_test.go]: amended the changes with the fix, ptal

**Comment by fabriziopandini** [comment]
@mkumatag thanks for the quick fix!
/approve
/lgtm

**Pull request merged**